### ISSUE TITLE
(small) bug in ROOTS

### ISSUE THREAD

**Issue by trefethen** [chebfun/chebfun#1576]
Nakatsukasa and Noferini have found the following bug.  The cubic 
```
f = chebfun('1e-10*x.^3 + x.^2 - 1e-12');
```
correctly has a negative minimum:
```
>> min(f)
ans =  -9.998668559774160e-13
```
It should have two roots, but it doesn't:
```
>> roots(f)
ans =
   Empty matrix: 0-by-1
```
See [their paper](http://www.keisu.t.u-tokyo.ac.jp/research/techrep/data/2014/METR14-29.pdf) for a discussion of what's going wrong.

**Comment by aaustin141** [comment]
We've visited this issue before:  @ajt60gaibb brought this up sometime during his tenure here.  The issue is that computing the roots with QZ is backward-stable provided the pencil is appropriately scaled, whereas using QR may not be.  See ticket #186 and the discussion there.  Back then, we decided to address this by creating a `qz` option for `roots` and hold off on switching to the QZ algorithm as default until we did more experimentation.  This option was incorporated into development in #192.  As a result, you can do
```
>> roots(f, 'qz')   
ans =
   1.0e-06 *
  -0.999940071244598
   0.999940071411132
>> f(ans)
ans =
     0
     0
```
to get the expected result.

I am therefore tempted to close this issue as a duplicate of #186 unless someone wants to get some more data to support changing to using QZ by the default.  The theoretical argument in favor of the switch is obvious.  I think the main argument against switching is that these cases where `roots` falls into this trap are rare enough that it doesn't justify taking the 2-3x-or-so performance hit.  (This trade-off is mentioned on p. 4 of the paper linked to by @trefethen in the original post.

**Comment by trefethen** [comment]
Ah, sorry, I didn't remember about #186.  I will go ahead and close this.  I do not recommend switching to QZ as the default, in the absence of evidence that our rootfinding is running into trouble in practice.  Thanks @aaustin141.

**Comment by ajt60gaibb** [comment]
I personally would like to see `chebfun/roots` use QZ but I think I am alone on that one...  

QZ is about 2.5x slower for large matrices, but `chebfun/roots` only computes eigenvalues of very small matrices. Last time I checked there was a 25% slow down in `chebfun/roots` with QZ, #186.  (This may have changed after #241.)
